### PR TITLE
Added tcp probes with prometheus blackbox-exporter

### PR DESCRIPTION
**Pull request by vonneudeck**
Unfortunately there is a bit SAP sauce.
Hopefully it won’t live long though.

### CODE REVIEW

**Comment by auhlig** [comment]
I'm not too happy if we add more SAP sauce, as it makes it harder to transition to stable/prometheus chart. Ok to get this in for now, but we need to clean up this up afterwards.
Please test in staging. Also the alert is missing.

**Comment by vonneudeck** [comment]
I would like to squash merge this now. Because I can’t meaningfully test the alert queries before rolling this out. The alerts will be done in a separate branch and PR afterwards.

File: system/kube-monitoring/charts/prometheus-frontend/README.md [hunk 1]
```diff
@@ -33,20 +33,26 @@ Region is an implicit label that is automatically set by Prometheus. Do not over
    * `openstack`
    * `datapath`
    * `kubernikus`
+   * `e2e`
```
Comment (auhlig) [inline system/kube-monitoring/charts/prometheus-frontend/README.md]: Why are you introducing the `e2e` tier? Currently we're using `tier=datapath, context=availability` for these tests.

File: system/kube-monitoring/charts/prometheus-collector/templates/_prometheus.yaml.tpl [hunk 2]
```diff
@@ -312,6 +312,29 @@ scrape_configs:
...
+  - source_labels: [__address__]
+    target_label: __param_target
+  - source_labels: [__param_target]
+    target_label: instance
+  - target_label: __address__
+    replacement: prober.{{ $region }}.cloud.sap
+  - target_label: region_probed_from
+    replacement: {{ $region }}
+{{ end }}
+
```
Comment (auhlig) [inline system/kube-monitoring/charts/prometheus-collector/templates/_prometheus.yaml.tpl]: ok for now, but for the next iteration: move to separate `extra_scrape_config`, just as we do in [kubernikus](https://github.com/sapcc/kubernikus/blob/master/charts/kubernikus-system/charts/prometheus/templates/config.yaml#L21-L23)

Comment (vonneudeck) [reply system/kube-monitoring/charts/prometheus-collector/templates/_prometheus.yaml.tpl]: Of course. I do not wish for this to stay there.

File: system/kube-monitoring/charts/prometheus-frontend/README.md (hunk 1 above)
Comment (auhlig) [reply system/kube-monitoring/charts/prometheus-frontend/README.md]: Can't follow based on your explanation. The tier label doesn't change the what/why, that's up to the respective test to do that. But okay if this was already decided.

**Comment by auhlig** [comment]
Please squash before merging

File: system/kube-monitoring/charts/prometheus-collector/templates/_prometheus.yaml.tpl [hunk 3]
```diff
@@ -312,6 +312,29 @@ scrape_configs:
...
 
+{{ range $region := .Values.global.regions }}
+- job_name: 'blackbox-tcp-{{ $region }}'
+  metrics_path: /probe
+  scrape_interval: 15s
+  params:
+    module: [tcp_connect]
+  static_configs:
+    - targets:
+      {{- range $.Values.blackbox_exporter.tcp_probe_targets }}
```
Comment (auhlig) [inline system/kube-monitoring/charts/prometheus-collector/templates/_prometheus.yaml.tpl]: this assumes there's a `blackbox_exporter.tcp_probe_targets` value. surround the whole job with with`{{ if blackbox_exporter.tcp_probe_targets }} ... {{ end }}`,

**Pull request merged**